### PR TITLE
Remove hardcoded MAX_DISK_CAPACITY_IN_BYTES and make it configurable.

### PR DESCRIPTION
**Pull request by ankagrawal**
With max disk capacity being hardcoded to 10TB currently, it's not possible to use disks with size greater than 10TB. This PR makes the max disk size configurable to allow for large disks.

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java [hunk 1]
```diff
@@ -266,6 +267,9 @@
   @Default("{}")
   public final JSONObject clustermapRecoveryTestHardwareLayout;
 
+  @Config(CLUSTERMAP_MAX_DISK_CAPACITY_IN_BYTES)
```
Comment (jsjtzyy) [inline ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java:270]: minor: java doc

File: ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java [hunk 2]
```diff
@@ -320,5 +324,7 @@ public ClusterMapConfig(VerifiableProperties verifiableProperties) {
         new JSONObject(verifiableProperties.getString("clustermap.recovery.test.hardware.layout", "{}"));
     clustermapRecoveryTestPartitionLayout =
         new JSONObject(verifiableProperties.getString("clustermap.recovery.test.partition.layout", "{}"));
+    clustermapMaxDiskCapacityInBytes =
+        verifiableProperties.getLong(CLUSTERMAP_MAX_DISK_CAPACITY_IN_BYTES, 10L * 1024 * 1024 * 1024 * 1024);
```
Comment (jsjtzyy) [inline ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java:328]: Could you change this to `getLongInRange`?  The parameters are:  min = 0, max = Long.MAX_VALUE, default = 20L * 1024 * 1024 * 1024 * 1024 (let's use 20TB as default value)

Comment (ankagrawal) [reply ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java]: done.

File: ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java (hunk 1 above)
Comment (ankagrawal) [reply ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java:273]: done.

**Pull request merged**